spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -298,6 +311,103 @@ public void deleteWhere(Filter[] filters) {
     }
   }
 
+  @Override
+  public StructType partitionSchema() {
+    if (lazyPartitionSchema == null) {
+      Table table = table();
+      PartitionsTable partitionsTable =
+          (PartitionsTable) MetadataTableUtils.createMetadataTableInstance(table, MetadataTableType.PARTITIONS);
+      Types.NestedField partition = partitionsTable.schema().findField("partition");
+      if (partition != null) {
+        this.lazyPartitionSchema = (StructType) SparkSchemaUtil.convert(partition.type());
```

Comment:
Just checking do we handle void transforms here correctly?

Reply:
> V1 Tables which have had a partition field removed leave this as a field in their spec
that is mean this is not suitable for use Types.NestedField. I am not sure if Igot it. Is there a related detailed doc?
I don't know if there are other ways to get partitions when use partition metadata table. I can use sparkTable.spec to instead it.

Reply:
Oh, I got it, when a partition field remove, it only mark spec transform void, so it may be return incorrectly. I am not sure if I understand is correct

Reply:
I think you are ok here since you are getting the type from the partitions metadata table. I would just make sure we have a test for it.

Reply:
> I think you are ok here since you are getting the type from the partitions metadata table. I would just make sure we have a test for it.


ok, and I have a question. for the  `partitionSchema` method.
```java
public StructType partitionSchema()
```
If I have a table, partitioned by dt, ht.  then I drop ht. So I think the method should only return partition dt. Is it right?

And in https://github.com/apache/iceberg/pull/3459#discussion_r749869700, the data from metadata partitions table will contain ht. So I think `listPartitionIdentifiers` doesn't conflict with `partitionSchema`?